### PR TITLE
Lynx: Long-Term Support testnet

### PR DESCRIPTION
**Pull request by cygnusv**
**Type of PR:**
- [ ] Bugfix
- [ ] Feature
- [X] Documentation
- [X] Other: updates to the registry

**Required reviews:** 
- [ ] 1
- [ ] 2
- [X] 3

**What this does:**
Adds the registry for the incoming Lynx testnet, a more stable environment to learn how to use NuCypher and integrate it in other apps.

**Issues fixed/closed:**
None

**Why it's needed:**
We're having increasingly demand for a better environment to learn how to integrate the NuCypher network with other apps, and the IBEX testnet has proven to be too unreliable for that (as it's being used mainly as a staking playground).
The Lynx testnet will hopefully be a step forward in facilitating adoption of NuCypher. 

There isn't a specific productdev issue for it, but it will contribute towards all issues focused in the current network offering.

**Notes for reviewers:**
There's really nothing much to look at. If you have the time, you can see the deployer account https://goerli.etherscan.io/address/0xecf5b563de623ce42bcedb8ec2188b995eb80a66, in case you want to double-check the data in the registry.

image

### CODE REVIEW

**Review by mswilkison — APPROVED** [review]

File: docs/source/guides/contribution_guide.rst [hunk 1]
```diff
@@ -201,8 +201,8 @@ Building Documentation
 
 .. note::
 
-  ``sphinx`` and ``sphinx_rtd_theme`` are non-standard dependencies that can be installed
-  by running ``pip install -e .[docs]`` from the project directory.
+  ``sphinx`` and ``sphinx_rtd_theme`` are non-standard dependencies required to build the documentation locally.
+  You can install them by running ``pip install sphinx sphinx_rtd_theme``.
```
Comment (derekpierre) [inline docs/source/guides/contribution_guide.rst:205]:
```suggestion
  You can install them by running ``pip install -e . -r docs-requirements.txt``.
```

File: newsfragments/2537.misc.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Introduces the Lynx testnet, a more stable environment to learn how to use NuCypher and integrate it to other apps.
```
Comment (derekpierre) [inline newsfragments/2537.misc.rst:1]:
```suggestion
Introduces the Lynx testnet, a more stable environment to learn how to use NuCypher and integrate it into other apps.
```

File: docs/source/guides/contribution_guide.rst (hunk 1 above)
Comment (cygnusv) [reply docs/source/guides/contribution_guide.rst:205]: Oh right, I forgot about the docs-requirements file.

**Review by derekpierre — APPROVED** [review]
🎸

File: nucypher/network/middleware.py [hunk 3]
```diff
@@ -145,8 +146,9 @@ class RestMiddleware:
     _client_class = NucypherMiddlewareClient
 
     TEACHER_NODES = {
-        'ibex': ('https://ibex.nucypher.network:9151',),
-        'mainnet': ('https://mainnet.nucypher.network:9151',),
+        NetworksInventory.MAINNET: ('https://mainnet.nucypher.network:9151',),
+        NetworksInventory.LYNX: ('https://lynx.nucypher.network:9151',),
+        NetworksInventory.IBEX: ('https://ibex.nucypher.network:9151',),
```
Comment (KPrasch) [inline nucypher/network/middleware.py:151]: :boom:

**Review by KPrasch — APPROVED** [review]
Staged and manually verified registry addresses :cowboy_hat_face:

**Pull request merged**